Modules/_asynciomodule.c

```diff
@@ -962,14 +962,18 @@ FutureObj_dealloc(PyObject *self)
 {
     FutureObj *fut = (FutureObj *)self;
 
+    PyObject_GC_UnTrack(self);
+
     if (Future_CheckExact(fut)) {
```

Comment:
but that would mean this (https://github.com/python/cpython/commit/34fae03cd6c9e304e02c571b3bf9e8df0cfe76be)  is wrong too then no? 

because it first does an UnTrack: https://github.com/python/cpython/commit/34fae03cd6c9e304e02c571b3bf9e8df0cfe76be#diff-c3cf251f16d5a03a9e7d4639f2d6f998L1113

With the same Track/Untrack pattern here:
https://github.com/python/cpython/commit/34fae03cd6c9e304e02c571b3bf9e8df0cfe76be#diff-c3cf251f16d5a03a9e7d4639f2d6f998R1129